kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriberServiceImpl.java

```diff
@@ -362,4 +372,28 @@ public void deleteAllowed(Subscriber subscriber) {
             subscriptionService.deletePreconditionCheck(subscription);
         }
     }
+
+    @Override
+    public void deactivateAllSubscriptionsForSubscriber(long callingNumber) {
+        LOGGER.info("Recieved Release Number {} for Deactivation.", callingNumber);
+        Subscriber subscriberByMsisdn = this.getSubscriber(callingNumber);
+        if (subscriberByMsisdn == null) {
+            LOGGER.info("Subscriber for msisdn {} is not found." , callingNumber);
+            throw new IllegalArgumentException(String.format(KilkariConstants.SUBSCRIBER_NOT_FOUND, callingNumber));
+        }
+        LOGGER.info("Found Subscriber for msisdn {} .", callingNumber);
+        int counter = 0;
+        for (Subscription subscription : subscriberByMsisdn.getAllSubscriptions()) {
+            if ((subscription.getStatus() == SubscriptionStatus.PENDING_ACTIVATION) || (subscription.getStatus() == SubscriptionStatus.ACTIVE) || (subscription.getStatus() == SubscriptionStatus.HOLD)) {
+                LOGGER.info("Deactivating Subscrition with Id {} for msisdn.", subscription.getSubscriptionId());
+                subscriptionService.deactivateSubscription(subscription, DeactivationReason.WEEKLY_CALLS_NOT_ANSWERED);
+                counter++;
+            }
+        }
+        LOGGER.info("Deactivated {} Subscritions for msisdn {}.", counter, callingNumber);
+        // Add callingNumber to WeeklyCallsNotAnsweredMsisdnRecord (tableName = "nms_weekly_calls_not_answered_msisdn")
+        weeklyCallsNotAnsweredMsisdnRecordDataService.create(new WeeklyCallsNotAnsweredMsisdnRecord(callingNumber, DateTime.now()));
```

Comment:
with the callingNumber set to unique on the domain object, what if the same deactivate api is called twice? what happens here?